src/components/NewCommentForm.tsx

```diff
@@ -2,7 +1,117 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import classNames from 'classnames';
+import React, { useEffect, useState } from 'react';
+import { postComment } from '../api/ClientFunctions';
+import { Comment } from '../types/Comment';
+
+interface Props {
+  postId: number;
+  commentsError: boolean;
+  setCommentsError: (value: boolean) => void;
+  setDoesFormExist: (value: boolean) => void;
+  setComments: React.Dispatch<React.SetStateAction<Comment[]>>;
+}
+
+export const NewCommentForm: React.FC<Props> = ({
+  postId,
+  commentsError,
+  setCommentsError,
+  setDoesFormExist,
+  setComments,
+}) => {
+  // #region states
+
+  const [isNameValid, setIsNameValid] = useState(true);
+  const [isEmailValid, setIsEmailValid] = useState(true);
+  const [isCommentValid, setIsCommentValid] = useState(true);
+  const [nameValue, setNameValue] = useState('');
+  const [emailValue, setEmailValue] = useState('');
+  const [commentValue, setCommentValue] = useState('');
+  const [isSubmitLoading, setIsSubmitLoading] = useState(false);
+
+  // #endregion
+  // #region handlers
+
+  const clearHandler = () => {
+    setNameValue('');
+    setEmailValue('');
+    setCommentValue('');
+    setIsNameValid(true);
+    setIsEmailValid(true);
+    setIsCommentValid(true);
+  };
+
+  const onSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    const IS_VALID =
+      nameValue.length > 0 && emailValue.length > 0 && commentValue.length > 0;
+
+    if (nameValue.length === 0) {
+      setIsNameValid(false);
+    }
+
+    if (emailValue.length === 0) {
+      setIsEmailValid(false);
+    }
+
+    if (commentValue.length === 0) {
+      setIsCommentValid(false);
+    }
+
+    if (!IS_VALID) {
```

Comment:
The form validation logic only checks for non-empty fields. Consider enhancing this by checking for valid email formats using a regular expression or a library function to improve user feedback.